Templates/make_jenkins_EC2-Agent-Windows-autoscale.tmplt.json

```diff
@@ -0,0 +1,1620 @@
+                                            "} -Verbose -ErrorAction Stop",
+                                            "\""
+                                        ]
+                                    ]
+                                },
+                                "waitAfterCompletion": "0"
+                            }
+                        }
+                    },
+                    "watchmaker-update": {
+                        "commands": {
+                            "10-watchmaker-update": {
+                                "command": {
+                                    "Fn::Join": [
+                                        "",
+                                        [
+                                            {
+                                                "Fn::FindInMap": [
+                                                    "ShellCommandMap",
+                                                    "powershell",
+                                                    "command"
+                                                ]
+                                            },
+                                            "-Command \"",
+                                            "Invoke-Command -ScriptBlock { ",
+                                            "$ErrorActionPreference = 'Stop'; ",
+                                            "c:\\cfn\\scripts\\watchmaker-install.ps1 -ErrorAction Stop -Verbose; ",
+                                            "watchmaker --log-level debug",
+                                            " --log-dir C:\\Watchmaker\\Logs",
+                                            " --salt-states None",
+                                            " --no-reboot",
+                                            {
+                                                "Fn::If": [
+                                                    "UseWamConfig",
+                                                    {
+                                                        "Fn::Join": [
+                                                            "",
+                                                            [
+                                                                " --config \\\"",
+                                                                {
+                                                                    "Ref": "WatchmakerConfig"
+                                                                },
+                                                                "\\\""
+                                                            ]
+                                                        ]
+                                                    },
+                                                    ""
+                                                ]
+                                            },
+                                            {
+                                       
```

Comment:
As noted previously, your "optional" `KeyName` isn't really optional: if you opted, above, to make `KeyPairName` actually optional, you're going to need to add an `Fn::If` construct meaning here. Otherwise, if `KeyPairName` is null, you're gonna have a bad time.

Reply:
Still an open issue...

Reply:
Resolved in f5d4c1fcf7e2435fd40b0886534e0624bfae96f7